src/olympia/devhub/views.py

```diff
@@ -1490,32 +1490,33 @@ def submit_finish(request, addon_id, addon):
     # Bounce to the versions page if they don't have any versions.
     if not addon.versions.exists():
         return redirect(addon.get_dev_url('versions'))
-    sp = addon.current_version.supported_platforms
-    is_platform_specific = sp != [amo.PLATFORM_ALL]
+    uploaded_version = addon.versions.latest()
+    supported_platforms = uploaded_version.supported_platforms
+    is_platform_specific = supported_platforms != [amo.PLATFORM_ALL]
 
     try:
         author = addon.authors.all()[0]
     except IndexError:
         # This should never happen.
         author = None
 
-    if author:
-        submitted_addons = (author.addons
-                            .exclude(status=amo.STATUS_NULL).count())
-        if submitted_addons == 1:
-            # We can use locale-prefixed URLs because the submitter probably
-            # speaks the same language by the time he/she reads the email.
-            context = {
-                'app': unicode(request.APP.pretty),
-                'detail_url': absolutify(addon.get_url_path()),
-                'version_url': absolutify(addon.get_dev_url('versions')),
-                'edit_url': absolutify(addon.get_dev_url('edit')),
-                'full_review': addon.status == amo.STATUS_NOMINATED
-            }
-            tasks.send_welcome_email.delay(addon.id, [author.email], context)
+    if (author and not author.addons.exclude(status=amo.STATUS_NULL)
```

Comment:
This test is really shaky -  we end up (currently) sending an email each time they submit a new add-on if their existing ones are rejected or deleted (the former would be `STATUS_NULL`; the latter excluded from `author.addons`).  
It's going to get worse when we have add-ons with all unlisted versions be STATUS_NULL permanently.

At the moment I can't think of anything better than doing a count of `not addon.is_incomplete()` (soon to be `.has_complete_metadata`) on the authors add-ons to filter incomplete submissions from rejected ones and only sending an email when they have no addons, or only incomplete ones.